pkg/stream/client.go

```diff
@@ -617,6 +617,12 @@ func (c *Client) BrokerForConsumer(stream string) (*Broker, error) {
 	if streamMetadata.responseCode != responseCodeOk {
 		return nil, lookErrorCode(streamMetadata.responseCode)
 	}
+
+	if streamMetadata.Leader == nil {
+		// in this case the leader is not ready yet
+		return nil, fmt.Errorf("leader not ready yet for stream: %s", stream)
```

Comment:
I think I like the idea of having a custom error type/instance here. And if we really need to include the stream ID in the error message, we can either wrap the well-defined error like `fmt.Errorf("error for stream %s: %w", LeaderNotReady)` that consumers can check with `errors.Is`, or create a custom error type that callers can check for with `errors.As` (I think I prefer the wrapping for simplicity).

Reply:
Added a specific Error